src/main/java/dk/cs/aau/huppaal/abstractions/Location.java

```diff
@@ -327,12 +332,17 @@ public void deserialize(final JsonObject json) {
         setInvariantY(json.getAsJsonPrimitive(INVARIANT_Y).getAsDouble());
     }
 
-
-
     @Override
     public String generateNearString() {
-        return "Location " + (!Strings.isNullOrEmpty(getNickname()) ? (getNickname() + " (" + getId() + ")") : getId());
+        if(parent == null)
```

Comment:
Can this be null at any point during execution?

Reply:
Yes. The constructor's parameter is not marked `@NotNull`, so it is technically possible. I think the help dialogue needs this 🤔

Reply:
I think you are right, maybe the help dialog should be using a different approach, but that is outside the scope of this PR